File: vllm_doctor/cli/migrate.py
from pathlib import Path

import typer

from vllm_doctor.cli import app
from vllm_doctor.config import load_config
from vllm_doctor.stores.migrate import run_migrations


@app.command()
def migrate(
    config_path: Path | None = typer.Option(
        None, "--config", "-c", help="Path to config file (default: vllm-doctor.toml)."
    ),
) -> None:
    """Run database migrations."""
    config = load_config(config_path)
    run_migrations(config.database.url)
    typer.echo("Database migrated successfully.")

File: src/cli/commands/migrate.rs
use std::path::PathBuf;

use crate::cli::stores::SqliteHistoryStore;

use super::{CommandResult, EXIT_ERROR, load_config_or_exit};

pub(super) async fn run(config: Option<PathBuf>) -> CommandResult {
    let config = load_config_or_exit(config.as_deref())?;
    SqliteHistoryStore::connect(&config.database.url)
        .await
        .map_err(|error| {
            eprintln!("Error: migration failed: {error}");
            EXIT_ERROR
        })?;
    println!("Database migrated successfully.");
    Ok(())
}
